gel/auth/email_password.py

```diff
@@ -0,0 +1,539 @@
+    status_code: int
+    message: str
+
+
+@dataclasses.dataclass
+class SignUpCompleteResponse:
+    verifier: str
+    token_data: td_mod.TokenData
+    identity_id: uuid.UUID
+
+
+@dataclasses.dataclass
+class SignUpVerificationRequiredResponse:
+    verifier: str
+    identity_id: Optional[uuid.UUID]
+
+
+@dataclasses.dataclass
+class SignUpFailedResponse(BaseServerFailedResponse):
+    verifier: str
+
+
+SignUpResponse = Union[
+    SignUpCompleteResponse,
+    SignUpVerificationRequiredResponse,
+    SignUpFailedResponse,
+]
+
+
+@dataclasses.dataclass
+class SignInCompleteResponse:
+    verifier: str
+    token_data: td_mod.TokenData
+    identity_id: uuid.UUID
+
+
+@dataclasses.dataclass
+class SignInVerificationRequiredResponse:
+    verifier: str
+    identity_id: Optional[uuid.UUID]
+
+
+@dataclasses.dataclass
+class SignInFailedResponse(BaseServerFailedResponse):
+    verifier: str
+
+
+SignInResponse = Union[
+    SignInCompleteResponse,
+    SignInVerificationRequiredResponse,
+    SignInFailedResponse,
+]
+
+
+@dataclasses.dataclass
+class EmailVerificationCompleteResponse:
+    token_data: td_mod.TokenData
+
+
+class EmailVerificationMissingProofResponse:
+    pass
+
+
+class EmailVerificationFailedResponse(BaseServerFailedResponse):
+    pass
+
+
+EmailVerificationResponse = Union[
+    EmailVerificationCompleteResponse,
+    EmailVerificationMissingProofResponse,
+    EmailVerificationFailedResponse,
+]
+
+
+@dataclasses.dataclass
+class SendPasswordResetEmailCompleteResponse:
+    verifier: str
+
+
+@dataclasses.dataclass
+class SendPasswordResetEmailFailedResponse(BaseServerFailedResponse):
+    verifier: str
+
+
+SendPasswordResetEmailResponse = Union[
+    SendPasswordResetEmailCompleteResponse,
+    SendPasswordResetEmailFailedResponse,
+]
+
+
+@dataclasses.dataclass
+class PasswordResetCompleteResponse:
+    token_data: td_mod.TokenData
+
+
+class PasswordResetMissingProofResponse:
+    pass
+
+
+class PasswordResetFailedResponse(BaseServerFailedResponse):
+    pass
+
+
+PasswordResetResponse = Union[
+    PasswordResetCompleteResponse,
+    PasswordResetMissingProofResponse,
+    PasswordResetFailedResponse,
+]
+
+
+C = TypeVar("C", bound=Union[httpx.Client, httpx.AsyncClient])
+
+
+class BaseEmailPassword(Generic[C]):
+    def __init__(self, *, connection_info: gel.ConnectionInfo, **kwargs):
+        self.provider = "builtin::local_emailpassword"
+        if "base_url" not in kwargs:
+            params = connection_info.params
+            scheme = "http" if params.tls_security == "insecure" else "https"
+            base_url = httpx.URL(
+                scheme=scheme,
+                host=connection_info.host,
+                port=connection_info.port,
+            )
+            kwargs["base_url"] = base_url.join(
+                f"branch/{params.branch}/ext/auth"
+            )
+        if "verify" not in kwargs:
+            kwargs["verify"] = connection_info.params.make_ssl_ctx()
```

Comment:
What does this part do?

Reply:
Oh, nevermind, i was getting this confused with email verification, but this is TLS stuff for the httpx client, got it.